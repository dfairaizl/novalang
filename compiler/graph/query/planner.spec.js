/* global describe, it, expect */

const Graph = require("../graph");

describe("Graph Query Planner", () => {
  describe("API", () => {
    it("plans basic queries", () => {
      const graph = new Graph();

      const q = graph.query();

      q.match({ type: "person" });

      expect(q.planner.ast).toMatchSnapshot();
    });

    it("plans basic queries to traverse nodes", () => {
      const graph = new Graph();

      const q = graph.query();

      q.match({ type: "person" }).outgoing("friends");

      expect(q.planner.ast).toMatchSnapshot();
    });

    it("plans basic queries with start node", () => {
      const graph = new Graph();

      const node = graph.addNode({ type: "temp" });

      const q = graph.query();

      q.begin(node).outgoing("friends");

      expect(q.planner.ast).toMatchSnapshot();
    });
  });
});
