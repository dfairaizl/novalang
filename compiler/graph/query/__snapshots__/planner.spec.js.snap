// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Graph Query Planner language plans basic queries 1`] = `
QueryAST {
  "head": MatchNode {
    "algorithm": "search",
    "children": Array [],
    "nodeFilters": Object {
      "type": "person",
    },
  },
}
`;

exports[`Graph Query Planner language plans basic queries to traverse nodes 1`] = `
QueryAST {
  "head": MatchNode {
    "algorithm": "search",
    "children": Array [
      TraversalNode {
        "algorithm": "bfs",
        "children": Array [],
        "edgeFilters": Array [
          "friends",
        ],
      },
    ],
    "nodeFilters": Object {
      "type": "person",
    },
  },
}
`;
